multisig/src/lib.rs

```diff
@@ -0,0 +1,376 @@
+                self.num_confirmations = num_confirmations;
+            }
+        };
+    }
+
+    /// Request to change number of confirmations.
+    pub fn set_num_confirmations(&mut self, num_confirmations: u32) -> RequestId {
+        assert_eq!(
+            env::current_account_id(),
+            env::signer_account_id(),
+            "Signer account must much given account"
+        );
+        self.add_request(MultiSigRequest::SetNumConfirmations { num_confirmations })
+    }
+
+    /// Request to transfer funds from this account to given account.
+    pub fn transfer(&mut self, account_id: AccountId, amount: U128) -> RequestId {
+        // TOOD: add safety margin for storage allocated balance.
+        assert_eq!(
+            env::current_account_id(),
+            env::signer_account_id(),
+            "Signer account must much given account"
+        );
+        assert!(
+            env::account_balance() >= amount.0,
+            "Not enough funds to initiate transfer"
+        );
+        self.add_request(MultiSigRequest::Transfer {
+            account_id,
+            amount: amount.0,
+        })
+    }
+
+    /// Request to add new access key to this contract.
+    /// Allows to extend allowed set of keys.
+    pub fn add_key(&mut self, public_key: Base58PublicKey) -> RequestId {
+        assert_eq!(
+            env::current_account_id(),
+            env::signer_account_id(),
+            "Signer account must much given account"
+        );
+        self.add_request(MultiSigRequest::AddKey {
+            public_key: public_key.into(),
+        })
+    }
+
+    /// Request to delete existing access key from this contract.
+    /// Allows to remove not used / lost / exposed key.
+    pub fn delete_key(&mut self, public_key: Base58PublicKey) -> RequestId {
+        assert_eq!(
+            env::current_account_id(),
+            env::signer_account_id(),
+            "Signer account must much given account"
+        );
+        self.add_request(MultiSigRequest::DeleteKey {
+            public_key: public_key.into(),
+        })
+    }
+
+    /// Request to call given contract's method with given argument / deposit / gas.
+    pub fn function_call(
+        &mut self,
+        contract_id: AccountId,
+        method_name: String,
+        args: Vec<u8>,
+        deposit: Balance,
+        gas: Gas,
+    ) -> RequestId {
+        assert_eq!(
+            env::current_account_id(),
+            env::signer_account_id(),
+            "Signer account must much given account"
+        );
+        self.add_request(MultiSigRequest::FunctionCall {
+            contract_id,
+            method_name,
+            args,
+            deposit,
+            gas,
+        })
+    }
+
+    /// Confirm given request with given signing key.
+    /// If with this, there has been enough confirmation, a promise with request will be scheduled.
+    pub fn confirm(&mut self, request_id: RequestId) {
```

Comment:
I think you should return `PromiseOrValue<bool>` here, when you execute the request, you return the promise. When adding a new confirmation, return `true`.